gusto/recovery/recovery_options.py

```diff
@@ -0,0 +1,67 @@
+
+from firedrake import (interval, FiniteElement, TensorProductElement, FunctionSpace)
+from gusto.core.function_spaces import DeRhamComplex
+from gusto.core.configuration import RecoveryOptions
+
+
+class RecoverySpaces():
+    """
+    Finds or builds necessary spaces to carry out recovery transport for lowest
+    and mixed order domains (0,0), (0,1) and  (1,0)
+    """
+
+    def __init__(self, domain):
+
+        family = domain.family
+        self.domain = domain
+        self.mesh = domain.mesh
+        # Need spaces from current deRham and a higher order deRham
+        self.de_Rham = DeRhamComplex(self.mesh, family,
+                                     horizontal_degree=1,
+                                     vertical_degree=1,
+                                     complex_name='recovery_de_Rham')
+
+        self.extruded_mesh = hasattr(self.mesh, "_base_mesh")
+        if self.extruded_mesh:
+            self.cell = self.mesh._base_mesh.ufl_cell().cellname()
+            self.build_theta_options()
+        else:
+            self.cell = self.mesh.ufl_cell().cellname()
+
+        self.build_DG_options()
+        self.build_HDiv_options()
+
+    def build_DG_options(self):
```

Comment:
Does this need to be its own function? Or could it be part of the __init__ method?

Reply:
I don't have particularly strong feelings one way or another about this. I wrote it this way as I found it easier to keep track of the different options when coding it up. I'm happy for it to be in the ini